mlmicrophysics/models.py

```diff
@@ -98,11 +106,11 @@ def fit(self, x, y, xv=None, yv=None):
         self.build_neural_network(inputs, outputs)
         self.model.summary()
         if self.classifier:
-            y_labels = np.unique(y)
-            y_class = np.zeros((y.shape[0], y_labels.size), dtype=np.int32)
+            self.y_labels = np.unique(y)
```

Comment:
y_labels and y_labels_val should have `self` removed from the front. Defining them as `self.y_labels` and `self.y_labels_val` retains an extra copy of the y data in memory within the DenseNeuralNetwork object after training rather than garbage collecting it once the fit method finishes.

Reply:
So, looking at this, I removed self.y_labels_val but kept self.y_labels because it's referenced in a method (?) in line 155. Let me know if you think there's a better way to deal with this.